src/commands/pull.js

```diff
@@ -0,0 +1,118 @@
+        const localConfig = await getLocalConfigOrThrow();
+        const userInfo = await getLocalUserInfo();
+        const apifyClient = await getLoggedClientOrThrow();
+        const cwd = process.cwd();
+
+        const isActorAutomaticallyDetected = !args?.actorId;
+        const usernameOrId = userInfo.username || userInfo.id;
+
+        const actorId = args?.actorId || `${usernameOrId}/${localConfig.name}`;
+
+        const actor = await apifyClient.actor(actorId).get();
+        if (!actor) throw new Error(`Cannot find actor with ID '${actorId}' in your account.`);
+
+        const { name, versions } = actor;
+
+        const latestVersion = versions.reduce((match, curr) => {
+            if (semverGt(`${curr.versionNumber}.0`, `${match.versionNumber}.0`)) return curr;
+            return match;
+        });
+
+        const dirpath = isActorAutomaticallyDetected ? cwd : path.join(cwd, name);
+        fs.mkdirSync(dirpath, { recursive: true }, null);
+
+        if (!isActorAutomaticallyDetected && !(fs.readdirSync(dirpath).length === 0)) {
+            error(`Directory ${dirpath} is not empty. Please empty it or choose another directory.`);
+            return;
+        }
+
+        let isPullSuccessful = false;
+
+        switch (latestVersion.sourceType) {
+            case 'TARBALL': {
+                await extractGitHubZip(latestVersion.tarballUrl, dirpath);
+                isPullSuccessful = true;
+                break;
+            }
+            case 'SOURCE_FILES': {
+                const { sourceFiles } = latestVersion;
+                for (const file of sourceFiles) {
+                    const folderPath = file.folder ? file.folder : path.dirname(file.name);
+                    fs.mkdirSync(`${dirpath}/${folderPath}`, { recursive: true }, null);
+
+                    if (!file.folder) {
+                        const fileContent = file.format === 'BASE64' ? Buffer.from(file.content, 'base64') : file.content;
+                        fs.writeFileSync(`${dirpath}/${file.name}`, fileContent);
+                    }
+                }
+                isPullSuccessful = true;
+                break;
+            }
+            case 'GIT_REPO': {
+                // e.g. https://github.com/jakubbalada/Datasety.git#master:RejstrikPolitickychStran
+                const { gitRepoUrl } = latestVersion;
+                const [repoUrl, branchDirPart] = gitRepoUrl.split('#');
+
+                let branch;
+                let dir;
+                if (branchDirPart) [branch, dir] = branchDirPart.split(':');
+                let branchDirRepoUrl = repoUrl;
+                if (dir) branchDirRepoUrl += `/${dir}`;
+                if (branch) branchDirRepoUrl += `#${branch}`;
+
+                const emitter = tiged(branchDirRepoUrl);
+                try {
+                    await emitter.clone(dirpath);
+                } catch (err) {
+                    error(`Failed to pull actor from ${gitRepoUrl}.`);
```

Comment:
Maybe add error message from err otherwise user do not know why cannot clone. Massage can help.

Reply:
```suggestion
                    error(`Failed to pull Actor from ${gitRepoUrl}.`);
```